File: CBuilder/ImapClient/NewFolderDlg.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "NewFolderDlg.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TNewFolderDialog *NewFolderDialog;
//---------------------------------------------------------------------------
__fastcall TNewFolderDialog::TNewFolderDialog(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TNewFolderDialog::FormCloseQuery(TObject *Sender,
      bool &CanClose)
{
  CanClose = (ModalResult != mrOk) || (edtName->Text != "");
  if (!CanClose) {
    ShowMessage("The value can not be empty");
  }
}
//---------------------------------------------------------------------------

File: Delphi/ImapClient/NewFolderDlg.pas
unit NewFolderDlg;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls;

type
  TNewFolderDialog = class(TForm)
    Label1: TLabel;
    edtName: TEdit;
    cbIsSubfolder: TCheckBox;
    Button1: TButton;
    Button2: TButton;
    procedure FormCloseQuery(Sender: TObject; var CanClose: Boolean);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  NewFolderDialog: TNewFolderDialog;

implementation

{$R *.dfm}

procedure TNewFolderDialog.FormCloseQuery(Sender: TObject; var CanClose: Boolean);
begin
  CanClose := (ModalResult <> mrOk) or (edtName.Text <> '');
  if not CanClose then
  begin
    ShowMessage('The value can not be empty');
  end;
end;

end.
